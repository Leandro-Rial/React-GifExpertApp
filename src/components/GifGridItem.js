import React from 'react';
import PropTypes from 'prop-types';

const GifGridItem = ({ title, url }) => {
    return (
        <div className="card animate__animated animate__fadeInLeft">
            <img src={url} alt={title} />
            <h2>{title}</h2>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}

export default GifGridItem
