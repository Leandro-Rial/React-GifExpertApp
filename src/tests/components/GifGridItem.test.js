import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem";

describe('Pruebas GifGridItem', () => {

    const title = "Titulo";
    const url = "http://localhost/test.png"

    let wrapper = shallow(<GifGridItem title={title} url={url} />)

    beforeEach(() => {
        wrapper = shallow(<GifGridItem title={title} url={url} />)
    })

    test('debe de mostrar <GifGridItem /> correctamente', () => {
        
        expect(wrapper).toMatchSnapshot()

    });

    test('debe de tener un parrafo con el title', () => {
        
        const titleGif = wrapper.find("h2").text().trim()

        expect(titleGif).toBe(title)

    });

    test('debe de tener la imagen igual al url y al alt de los props', () => {
        
        const imgGif = wrapper.find("img");

        expect(imgGif.prop("src")).toBe(url);
        expect(imgGif.prop("alt")).toBe(title);

    });

    test('debe de tener animate__fadeInLeft', () => {
        
        const div = wrapper.find("div");

        expect(div.hasClass("animate__fadeInLeft")).toBe(true);

    });
});