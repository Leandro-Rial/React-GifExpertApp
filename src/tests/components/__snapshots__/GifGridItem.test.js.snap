// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas GifGridItem debe de mostrar <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__fadeInLeft"
>
  <img
    alt="Titulo"
    src="http://localhost/test.png"
  />
  <h2>
    Titulo
  </h2>
</div>
`;
